plugins/woocommerce/includes/gateways/bacs/class-wc-gateway-bacs.php

```diff
@@ -282,16 +284,17 @@ public function email_instructions( $order, $sent_to_admin, $plain_text = false
 		 * Filter the email instructions order status.
 		 *
 		 * @since 7.4
+		 *
 		 * @param string $terms The order status.
 		 * @param object $order The order object.
 		 */
-		if ( ! $sent_to_admin && 'bacs' === $order->get_payment_method() && $order->has_status( apply_filters( 'woocommerce_bacs_email_instructions_order_status', 'on-hold', $order ) ) ) {
+		$instructions_order_status = apply_filters( 'woocommerce_bacs_email_instructions_order_status', OrderStatus::ON_HOLD, $order );
+		if ( ! $sent_to_admin && 'bacs' === $order->get_payment_method() && $order->has_status( $instructions_order_status ) ) {
```

Comment:
Filter invocation frequency change.

Reply:
Fixed in https://github.com/woocommerce/woocommerce/pull/51922/commits/9071d081dfe753615780eec18284c19023d255c9